### PR TITLE
setup test environment

### LINKED ISSUE #124: Setup test environment
## Feature Title

To test code that interacts with DB, it needs the mock DB. In this project, we use the docker image of DynamoDB as a mock DB. Therefore we need to change the way of the test in the GitHub Actions from using Nodejs to using docker-compose.

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Updated the test environment from the node base to the docker base.

## Review points

I hope this works well.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -8,15 +8,8 @@ jobs:
...
-      - name: Install Node.js
-        uses: actions/setup-node@v4
-        with:
-          node-version: 18
-      - name: Install Dependencies
-        run: npm ci
-        working-directory: ./backend/main
       - name: Run Tests
-        run: npm test
+        run: docker compose up app --exit-code-from app
```
Comment (hitohata) [inline .github/workflows/test.yml:12]: This option is needed to stop running docker after a test is finished.

https://docs.docker.com/engine/reference/commandline/compose_up/

File: backend/main/docker-compose.yml [hunk 2]
```diff
@@ -5,9 +5,10 @@ services:
     volumes:
       - .:/code
     working_dir: /code
+    command: bash -c "npm ci && npm run test:run"
     depends_on:
-      - dynamo-db
-      - initializer
+      initializer:
+        condition: service_completed_successfully
```
Comment (hitohata) [inline backend/main/docker-compose.yml:11]: This condition means that this container waits until the initializing Table is complete.

https://docs.docker.com/compose/compose-file/05-services/#long-syntax-1

File: backend/main/package.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "typedoc": "typedoc",
     "typedoc:watch": "typedoc --watch",
-    "test": "vitest"
+    "test": "vitest",
+    "test:run": "vitest run"
```
Comment (hitohata) [inline backend/main/package.json:10]: The default mode of the vitest is watch mode. The "run" commands prevent it and after finishing the test, stop the execution.

https://vitest.dev/guide/cli#vitest-run

File: backend/main/docker-compose.yml (hunk 2 above)
Comment (taniguchiiqqq) [reply backend/main/docker-compose.yml:11]: Wow, cool

**Review by taniguchiiqqq — APPROVED** [review]
Thank you! I'm impressed to know some tips in this PR 🙌
I verified that the initializer works properly when I run docker-compose twice.

**Pull request merged**